build.sbt

```diff
@@ -63,9 +63,9 @@ val sparkTestingVersion = settingKey[String]("Spark testing base version without
 // Core (non-JNI bits)
 
 lazy val core = (project in file("core")) // regular scala code with @native methods
-  .settings(javah / target := (native / nativeCompile / sourceDirectory).value / "include")
   .dependsOn(native % Runtime)
-  .dependsOn(native)
+  .settings(javah / target := (native / nativeCompile / sourceDirectory).value / "include")
+  .settings(sbtJniCoreScope := Compile)
```

Comment:
@holdenk are you sure you need it in Compile scope? 

What this setting does, it brings the sbtjni-core into the app rutime, it is neccessary for the Scala3 support (it does not support macro annoations). 

Check this out: https://github.com/sbt/sbt-jni#usage-example-scala-3x--scala-2x

If you only plan to use annotations, you don't need it to be set to Compile! (By default it is in the Provided scope)